app/customer.py

```diff
@@ -0,0 +1,35 @@
+import dataclasses
+import datetime
+
+
+@dataclasses.dataclass
+class Customer:
+    name: str
+    product_cart: dict
+    location: list
+    money: float
+    car: dict
+
+    def the_cheapest_trip_to_the_store(self, shops: dict) -> str:
+        result = "Outskirts Shop"
+        cost = self.money
+        for shop_name, shop_value in shops.items():
+            if shop_value < cost:
+                cost = shop_value
+                result = shop_name
+        return result
+
+    def print_check(self, cheapest_shop: dict, total_cost: float) -> None:
+        date_of_purchase = \
+            datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")

```

Comment:
move to one line. It is better to import:
`from datetime import datetime`

Reply:
if i import from datetime import datetime, pytest fails
In test_main MagicMock(wrap=datetime.datetime)
because of this, I think pytest fails